### DISCUSSION TITLE
Is Nginx (or Similar) Needed When Self Hosting?

### DISCUSSION THREAD

**Discussion by darrylmorley** [vercel/next.js#29376] [Help] [answered]
Hi,

I'm going to be deploying using docker and I'm wondering, is Nginx (or similar) actually needed? And, if so, why? NextJs runs it's own server which can be accessed from the web? Would anybody mind delving into the advantages/disadvantages of using an additional web server?

**Comment by kachkaev** [comment, marked as answer]
It depends on how you build the Next.js app. If you use [static html export](https://nextjs.org/docs/advanced-features/static-html-export) feature, then you end up with a bunch of generated files that need to be served via Nginx or alike. Server load will be tiny, but you won’t get server-side-rendering or API routes.

If you only run `next build` (and not `next build && next export`), the result needs to be served with Node.js running inside your container. This includes static assets, so you won’t need Nginx. In both cases it’s best to place your container behind a reverse proxy of some sort but that’s irrelevant to Next.js as such.

**Reply by darrylmorley** [reply]
@kachkaev Thanks for your response!
